sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -86,13 +86,13 @@ template <typename AllocatorT> class SYCLMemObjT : public SYCLMemObjI {
 
     RT::PiMem Mem = pi::cast<RT::PiMem>(MInteropMemObject);
     RT::PiContext Context = nullptr;
-    PI_CALL(RT::piMemGetInfo, Mem, CL_MEM_CONTEXT, sizeof(Context), &Context,
-            nullptr);
+    RT::piCheckResult(PI_CALL_NOCHECK(piMemGetInfo)(
```

Comment:
This looks...kind of strange: what so specific is being done by `piCheckResult` that cannot be done by regular `PI_CALL`?

Reply:
I don't think we even need piCheckResult. We should write error specific result checking for each pi invocation.

Reply:
@romanovvlad Please confirm if i understood you correct/addressed your concern.